llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1927,6 +1921,24 @@ void VPScalarCastRecipe ::print(raw_ostream &O, const Twine &Indent,
 }
 #endif
 
+void VPScalarStoreRecipe::execute(VPTransformState &State) {
+  IRBuilderBase &Builder = State.Builder;
+  Value *StoredVal = State.get(getStoredVal(), 0, /*IsScalar*/ true);
```

Comment:
if it is sunk outside, shouldn't it store the last lane of the last unrolled iteration? as is, it would generate incorrect code when interleaving with other users?

Or alternatively enforce that the operand are also defined outside any vector region?

Reply:
ee8ce9f65a27313977403fd74f84292bac40d7ea
Check operands in the ::execute function. 
Initially, I attempted to check the operand in the constructor, but given the current application, the stored value is only correctly assigned in FixReduction.